druid/src/widget/svg.rs

```diff
@@ -427,8 +446,13 @@ fn transform_to_affine(t: usvg::Transform) -> Affine {
     Affine::new([t.a, t.b, t.c, t.d, t.e, t.f])
 }
 
-fn color_from_svg(c: usvg::Color, opacity: usvg::Opacity) -> Color {
-    Color::rgb8(c.red, c.green, c.blue).with_alpha(opacity.value())
+fn color_from_svg(c: usvg::Color, opacity: usvg::Opacity, default: &Color) -> Color {
+    let computed = Color::rgb8(c.red, c.green, c.blue);
+    match computed {
+        Color::BLACK => default.clone(), // usvg uses black as its default.
```

Comment:
I'm confused by this, because it looks as though you'll end up replacing everything black by the default color, even if black is specified in the svg. Does `usvg` not give a way to tell the difference between, say, `<rect x="10" y="10" width="100" height="100" stroke="black"/>` and `<rect x="10" y="10" width="100" height="100"/>`?

Also, you called the method `default_stroke_color`, but it seems to affect fills also?

Reply:
> Does usvg not give a way to tell the difference between, say, <rect x="10" y="10" width="100" height="100" stroke="black"/>

correct. this is less than an ideal solution.

> Also, you called the method default_stroke_color, but it seems to affect fills also?

true, happy to change the name, its basically the default painting color I guess. =)

Reply:
for [reference](https://github.com/RazrFalcon/resvg/blob/d2ead624b23c995a966251531e734aed95c368a5/usvg/src/style.rs#L157)

Reply:
Hm, is it possible for you to work around this another way, for example by just changing the svg files? I'm not super keen to add a feature that we know we can't implement correctly...

Reply:
not really no. I'm loading applications icons, which I don't have direct color over.

I don't think it'd be that hard to work with resvg to fix the underlying problem to allow us to remove the black check. I'm just going to have minimal free time soon so I can't spear head it.

this feature is also opt in. it has no change in current behavior unless you explicitly set the values.

does your objection disappear if usvg receives a fix?

Reply:
If this really only substituted missing colors, I wouldn't have any objections.

Reply:
I also need this, a solution would be to let user provide the `color_from_svg` function